### ISSUE TITLE
false negative when overriding method

### ISSUE THREAD

**Issue by theosotr** [typetools/checker-framework#6864]
### Command

```
javac -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
-processor org.checkerframework.checker.nullness.NullnessChecker \
-cp checker-framework-3.47.0/checker/dist/checker.jar Test.java
```

### File

```java
import org.checkerframework.checker.nullness.qual.*;
import java.util.*;

class A {
    <T extends List<Integer>> T m(T x) {
        return x;
    }
}

class B extends A {
    <T extends List<@Nullable Integer>> T m(T x) {
        x.add(null);
        return x;
    }
}


public class Test {
    public static void main(String[] args) {
        A x = new B();
        List<Integer> y = new LinkedList<>();
        x.m(y).get(0).toString();
    }
}
```

### Actual behavior

The code passes the checks, but there's a NPE at runtime

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Integer.toString()" because the return value of "java.util.List.get(int)" is null
        at Test.main(Test.java:22)

```

### Expected behavior

The code should have been rejected because class B changes the signature of the overriden method.

**Comment by Suvrat1629** [comment]
Hi @mernst @smillst ,

I have developed a fix for the unsoundness reported in this issue. I investigated through the `BaseTypeVisitor` and saw that there is no validation for method-level type parameters. I would like to add that by adding a method -  `checkTypeParameters()` to `BaseTypeVisitor.OverrideChecker`. I also would like to introduced a new diagnostic key, `override.typeparam`, to provide clear feedback when bounds are incompatible.

Does this approach sound good to you? If so, I’d like to open a PR for this.
